### ISSUE TITLE
Jetstream fails to send message to consumer(s)

### ISSUE THREAD

**Issue by avalchev94** [nats-io/nats-server#3598]
## Defect

Hello, I have a JetStream data folder(attached), which has couple of streams, messages and consumers created in it. As the title suggest, one(maybe more) of those consumers are not receiving messages sent to their streams. I am not really sure how it ended like that, however I know that it happened after upgrading the services(golang) that are using the jetstream server. If needed, I will give more context/data.

#### Versions of `nats-server` and affected client libraries used:
nats-server v2.6.6
nats-client: v1.11.0 (golang)

#### OS/Container environment:
It was initially reproduced on Windows, however using the jetstream files, it's also reproducible on MacOS.

#### Steps or code to reproduce the issue:
Before writing the steps, let me clarify:
- issue is reproducible only if you use the attached jetstream data folder;
- issue is reproducible only with **profile** stream and some of it's consumers(ex. **profiles_watcher**);
- if you delete the jetstream folder and re-create all streams and consumers, everything works correctly;

To reproduce:
1. Start nats server using the attached jestream folder:
`nats-server -js -DV -sd /path/to/jestream/folder`
2. Start **profiles_watcher** consumer:
`nats consumer sub profile profiles_watcher`
3. Send message to **profile**  stream:
`nats pub profile.create create-message` 

#### Expected result:
The consumer started from step 2 to receive the message

#### Actual result:
The consumer from step 2 does not receive the message

#### Additional info:
If you repeat steps 2 and 3, but using **device_config** stream and "configs_watcher" consumer, the message will be received:
```
nats consumer sub device_config configs_watcher

>> subj: device_config.create / tries: 1 / cons seq: 38 / str seq: 38 / pending: 0
message123

nats pub device_config.create "message123"
```

[nats.zip](https://github.com/nats-io/nats-server/files/9921605/nats.zip)

**Comment by ripienaar** [comment]
```
nats-server v2.6.6
```

Is relatively speaking ancient with 100s of bug fixes since, you should try upgrading it's impossible to say if this bug is fixed since.

**Comment by avalchev94** [comment]
> ```
> nats-server v2.6.6
> ```
> 
> 
>      
>   
> 
> Is relatively speaking ancient with 100s of bug fixes since, you should try upgrading it's impossible to say if this bug is fixed since.

I agree, but can't we, at least, try to find what's wrong with the attached jetstream storage files? I loaded those files on v2.9.5 and the issue was still there.

**Comment by kozlovic** [comment]
Having a look now.

**Comment by kozlovic** [comment]
@avalchev94 Here is a `nats stream info profile`, followed by the `nats consumer report profile`. I am omitting some part of the stream info to bring attention to the reason why the consumers do not receive messages:
```
State:

             Messages: 13
                Bytes: 2.5 KiB
             FirstSeq: 4 @ 2022-11-02T11:32:37 UTC
              LastSeq: 16 @ 2022-11-02T14:23:44 UTC
     Active Consumers: 4
   Number of Subjects: 1

iMacSynadia:~ ivan$ nats c report profile
╭─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                              Consumer report for profile with 4 consumers                                                               │
├────────────────────────┬──────┬────────────┬──────────┬─────────────┬─────────────┬─────────────┬───────────┬───────────────────────────────────────────────────────────┤
│ Consumer               │ Mode │ Ack Policy │ Ack Wait │ Ack Pending │ Redelivered │ Unprocessed │ Ack Floor │ Cluster                                                   │
├────────────────────────┼──────┼────────────┼──────────┼─────────────┼─────────────┼─────────────┼───────────┼───────────────────────────────────────────────────────────┤
│ durable-profiles       │ Push │ Explicit   │ 30.00s   │ 0           │ 0           │ 7 / 53%     │ 0         │ NC5EAVNMXRGJBDFOWZCGQZ6B3BBGQC6ISXZWGSYDTSF2ZOLHSTHLZBGZ* │
│ files_profiles_watcher │ Push │ Explicit   │ 30.00s   │ 0           │ 0           │ 0           │ 130       │ NC5EAVNMXRGJBDFOWZCGQZ6B3BBGQC6ISXZWGSYDTSF2ZOLHSTHLZBGZ* │
│ profiles_watcher       │ Push │ Explicit   │ 30.00s   │ 0           │ 0           │ 0           │ 130       │ NC5EAVNMXRGJBDFOWZCGQZ6B3BBGQC6ISXZWGSYDTSF2ZOLHSTHLZBGZ* │
│ profiles_watcher_acs   │ Push │ Explicit   │ 30.00s   │ 0           │ 0           │ 0           │ 130       │ NC5EAVNMXRGJBDFOWZCGQZ6B3BBGQC6ISXZWGSYDTSF2ZOLHSTHLZBGZ* │
╰────────────────────────┴──────┴────────────┴──────────┴─────────────┴─────────────┴─────────────┴───────────┴───────────────────────────────────────────────────────────╯
```
Notice of the last sequence in the stream: 16. But look at the AckFloor of the consumers in question: 130. These consumers would receive nothing until the stream sequence goes past that point.

If you send more messages, I believe that the consumers will be unblocked (and verified that myself by using `--count 200` in the nats pub command)

As @ripienaar mentioned, there have been so many fixes since this version that the state could have been "corrupted" in ways that just replacing with v2.9.5 does not immediately solve the issue. However, v2.9.5 may not get you in that situation to start with.

**Comment by avalchev94** [comment]
@kozlovic Thanks for the insight, it is really helpful. Just a guess here, could the problem happens because of the maximum age policy(7 days)? Restarting nats triggered some startup cleanup procedure that deleted all messages for **profile** stream and, as a result, all new messages from the stream started from sequence id - 1. 

I might try reproduce that with smaller maximum age policy tomorrow.

**Comment by kozlovic** [comment]
@avalchev94 Yes, I believe we had this issue in the past. But not sure which exact version fixed that.

**Comment by avalchev94** [comment]
Maybe that one - https://github.com/nats-io/nats-server/pull/2753

Will try to reproduce tomorrow and will give more info and, hopefully, close the ticket.

**Comment by kozlovic** [comment]
@avalchev94 If that is the case, it would have been fixed in v2.7.0, which is a newer release that you are using, so definitively possible.

**Comment by avalchev94** [comment]
Confirmed, the issue is caused by ungraceful exit of NATS server. Tested with 2.9.5 and it seems to work correctly.
